docs/sphinx/cookbook/resource_aware_pool.rst

```diff
@@ -0,0 +1,150 @@
+.. _resource_aware_pool:
+
+=============================
+Creating a Resource Aware Pool
+=============================
+
+This recipe describes how to create and use an Umpire ``ResourceAwarePool``. This pool is somewhat advanced
+so we also provide a bit of background on Camp resources which are used to track resources and events.
+
+Camp Resources
+--------------
+
+Umpire uses `Camp <https://github.com/LLNL/camp>`_ resources to keep track of "streams of execution". A single "stream of execution" 
+on the device corresponds to a single Camp device resource (e.g. a single cuda stream). 
+Similarly, when we are executing on the host, this
+corresponds to a separate "stream of execution" and therefore a separate Camp host resource. 
+
+Typically, we deal with multiple Camp resources. This includes a single resource for the host and 
+one or more for the device, depending on how many (cuda, hip, etc.) streams we have in use. 
+While we can have multiple camp resources for the device (e.g. multiple cuda streams), 
+we can only have one resource for the host because the host only has one stream of execution. 
+Since we are dealing with Camp resources, we call this pool strategy the ``ResourceAwarePool``.
+
+Throughout the rest of this documentation page, we will use a "camp resource" to refer to a "stream of
+execution". If the camp resource is on the device, then we are referring to a device stream such 
+as a cuda stream or hip stream.
+
+Using a Single Resource
+-----------------------
+
+Umpire's strategies such as ``QuickPool`` and ``DynamicPoolList`` work very well
+on the device when we are dealing with a single camp device resource. In the figure below, we have
+the host resource which allocates memory (a\ :sub:`1`\), uses the memory in a kernel (k\ :sub:`1`\1), then schedules
+a deallocate (d``_``1). Then, the host immidiately reuses that memory for a different kernel (k``_``2).
```

Comment:
```suggestion
a deallocate (d\ :sub:`1`\). Then, the host immidiately reuses that memory for a different kernel (k\ :sub:`2`\).
```

Two more subscript instances.

Reply:
Haha we were essentially editing the file at the same time - I made the change, thanks!